### PR TITLE
4B length

### LINKED ISSUE #129: Shorten nonce length for SCID
Way back in #33 , @huitema said that his three-pass algorithm meant that

> ...it should also be possible to make the nonce bit shorter than the original spec, because we are no more relying on the randomness of the nonce value. Each server could set the nonce to some kind of sequence number, incremented at each CID allocation. The size of the number should be enough to cover all allocations during a CID encryption key epoch, instead of twice that to cover the birthday paradox if using random allocations.

But we never really did that: the nonce is in the range 8..16 octets, which is inherited from the original single-pass algorithm.

@ianswett asked if we could shorten the minimum nonce to 32 bits to reduce overall CID length; if it were, we could support 2^24 servers with an 8-byte CID!

This is compelling and we should definitely shorten the nonce to some value. But 32 bits means you need to roll over the keys for every 4 billion CIDs a server issues; given that some QUIC implementations are quite profligate issuing CIDs, is that enough?

Even another byte would get us to a trillion CIDs while supporting up to 2^16 servers in 8 octets. On the other hand, the config agent can always pick a longer nonce length if the key lifetime is too short. This is how much we want to allow people to do something potentially dumb, so it's a good spot for discussion.

### PR DESCRIPTION
**Pull request by martinduke**
Fixes #129.

After a lot of thinking about nonce reuse, I added somewhat related text about stateless reset token key management.

### CODE REVIEW

File: draft-ietf-quic-load-balancers.md [hunk 1]
```diff
@@ -1297,13 +1297,21 @@ that share the same cryptographic context for Stateless Reset tokens. As QUIC-LB
...
+to generate correct tokens for connections using either type of CID.
+
 ## Connection ID Entropy
 
 The Stream Cipher and Block Cipher algorithms need to generate different cipher
 text for each generated Connection ID instance to protect the Server ID. To
-do so, at least four octets of the Block Cipher CID and at least eight octets
-of the Stream Cipher CID are reserved for a nonce that, if used only once, will
-result in unique cipher text for each Connection ID.
+do so, at least four octets of the CID are reserved for a nonce that, if used
```
Comment (ianswett) [inline draft-ietf-quic-load-balancers.md:1313]: This makes it look like you're reducing the minimum nonce size for both stream and block ciphers.  Is that correct?

**Review by ianswett — APPROVED** [review]
LG.  The intent is to reduce the minimum nonce length for both Stream and Block ciphers, correct?

File: draft-ietf-quic-load-balancers.md (hunk 1 above)
Comment (martinduke) [reply draft-ietf-quic-load-balancers.md:1313]: No, we're converging the minimum to 4 for all cases (it was already at 4 for BCCID)

**Review by huitema — APPROVED** [review]
Looks fine to me.

**Review by huitema — APPROVED** [review]
Let's do this.

**Review by nibanks — APPROVED** [review]
LGTM

**Pull request merged**